pkg/flare/archive.go

```diff
@@ -290,7 +290,7 @@ func writeStatusFile(tempDir, hostname string, data []byte) error {
 func addParentPerms(dirPath string, permsInfos permissionsInfos) {
 	parent := filepath.Dir(dirPath)
 	for parent != "." {
-		if parent == "/" {
+		if parent == "/" || parent == "C:\\" {
```

Comment:
Are we sure that there's no way this function could be used with `dirPath` outside of the `C:` drive?
I think the log file path can be overriden with the `log_file` attribute: https://github.com/DataDog/datadog-agent/blob/da5db0e8358eaa971343f234f7d7f5ddd8516468/pkg/config/config_template.yaml#L1114-L1118

Reply:
You're right, didn't think of that. An alternative would be to check if the same directory is encountered twice, but that could break if there are two directories with the exact same name in the hierarchy. Or I could set a max for the for loop, but that isn't really elegant.

Reply:
> An alternative would be to check if the same directory is encountered twice, but that could break if there are two directories with the exact same name in the hierarchy.

I don't think this should be possible, barring some weird symlink wizardry? For example, if your path is `C:\Temp\Log\Log\Test`, `parent` would have the following values:
```
C:\Temp\Log\Log
C:\Temp\Log
C:\Temp
C:\
C:\
```
so the parent and its parent would only have the same values when they're the root directory.

Reply:
Yeah you're right. I think I found a more elegant solution by using the Go `filepath.Separator` constant. I can just break on that, and it should work correctly on any OS.